tests/e2e/app/src/Main.js

```diff
@@ -0,0 +1,175 @@
+        Updates.clear();
+        Onyx.merge(ONYXKEYS.SESSION, {login: 'test@test.com'});
+        Onyx.merge(ONYXKEYS.RANDOM_NUMBER, {number: Date.now()});
+    };
+
+    const onLogOut = () => {
+        Onyx.clear([]);
+    };
+
+    const onFetchPokedex = () => {
+        Onyx.merge(ONYXKEYS.POKEDEX, Data.pokedex);
+    };
+
+    const onFetchSpaceData = (small) => {
+        const date = Date.now();
+
+        for (let i = 0; i <= (small ? 10 : 100); i++) {
+            Onyx.merge(
+                `${ONYXKEYS.COLLECTION.METEORITES}${date}${i}`,
+                Data.meteorites,
+            );
+            Onyx.merge(
+                `${ONYXKEYS.COLLECTION.ASTEROIDS}${date}${i}`,
+                Data.asteroids,
+            );
+        }
+    };
+
+    return (
+        <View style={styles.container}>
+            {isAuthenticated ? (
+                <View style={styles.container}>
+                    <Text>{props.session.login}</Text>
+                    <Button
+                        title="Log Out"
+                        onPress={onLogOut}
+                    />
+                    <View style={styles.containerButtons}>
+                        <Button
+                            title="Fetch Pokedex"
+                            testID="fetch-pokedex-data"
+                            onPress={onFetchPokedex}
+                        />
+                        <Button
+                            title="Fetch Space data"
+                            testID="fetch-space-data"
+                            onPress={() => onFetchSpaceData(false)}
+                        />
+                        <Button
+                            title="Fetch (small) Space data"
+                            testID="fetch-small-space-data"
+                            onPress={() => onFetchSpaceData(true)}
+                        />
+                    </View>
+                </View>
+            ) : (
+                <Button title="Log In" onPress={onLogIn} />
+            )}
+            <Text aria-label="leader">
+                {isLeader ? 'leader' : 'non-leader'}
+            </Text>
+            <Text aria-label="data-number">
+                {lodashGet(props, 'randomNumber.number', undefined)}
+            </Text>
+            <Text aria-label="data-pokedex">
+                {lodashGet(props, 'pokedex.pokemon.length', undefined)}
+            </Text>
+            <Text aria-label="data-meteorites" numberOfLines={10}>
+                {_.filter(_.map(Object.entries(props.allMeteorites || {}),
+                    ([key, value]) => (value ? `${key}-${value && value.length}` : undefined)), v => !!v)
+                    .join(',')}
+            </Text>
+            <Text aria-label="data-asteroids" numberOfLines={10}>
+                {_.filter(_.map(Object.entries(props.allAsteroids || {}),
+                    ([key, value]) => (value ? `${key}-${value && value.length}` : undefined)), v => !!v)
+                    .join(',')}
```

Comment:
Looks like it is identical. Can we create a function to re-use?

Reply:
I don't think it's worth the effort, it's just two entries, in a test app. Re-using would create more code and be more confusing